### PR TITLE
add copy transcript button

### PR DESCRIPTION
**Pull request by marndt93**
resolves: #3469
Adds a copy transcript button which copies game log to clipboard.
image

image

### CODE REVIEW

**Comment by tobymao** [comment]
how does this look on mobile?

File: assets/app/view/game/game_log.rb [hunk 1]
```diff
@@ -226,8 +228,36 @@ def render_date_banner(time)
         h('div.chatline', { style: { textAlign: :center } }, date)
       end
 
+      def copy_log_transcript
+        actions = @game.actions.to_h { |a| [a.id, a] }
+        logText = ''
```
Comment (tobymao) [inline assets/app/view/game/game_log.rb:233]: snake_case

**Comment by marndt93** [comment]
> 

what is the min screen size supported for mobile?

File: assets/app/view/game/game_log.rb (hunk 1 above)
Comment (tobymao) [reply assets/app/view/game/game_log.rb:233]: change this to String.new and do string << x or an array and do [].join()

**Comment by tobymao** [comment]
that's probably fine

File: assets/app/view/game/game_log.rb [hunk 2]
```diff
@@ -226,8 +228,36 @@ def render_date_banner(time)
...
+        logText = String.new
+        @game.log.map.group_by(&:action_id).flat_map do |action_id, entries|
+          action = actions[action_id]
+          entries.flat_map do |entry|
+            time = action&.created_at ? "[#{Time.at(action.created_at || Time.now).strftime('%R')}]" : ""
+            line = entry.message
+            if line.is_a?(Engine::Action::Message)  
+              sender = line.entity.name || line.user
+              line = "#{sender}: #{line.message}"
+            end
```
Comment (tobymao) [inline assets/app/view/game/game_log.rb:242]: log_text << z

Comment (marndt93) [reply assets/app/view/game/game_log.rb:242]: I do not know this syntax/ruby.  Is this code valid?
image
image

Comment (tobymao) [reply assets/app/view/game/game_log.rb:242]: oh poop, ok just use an array then and then finish with a join

Comment (marndt93) [reply assets/app/view/game/game_log.rb:242]: also, using String.new fails the linter:
image

File: assets/app/view/game/game_log.rb [hunk 3]
```diff
@@ -230,32 +230,32 @@ def render_date_banner(time)
...
+            time = action&.created_at ? "[#{Time.at(action.created_at || Time.now).strftime('%R')}]" : ''
             line = entry.message
-            if line.is_a?(Engine::Action::Message)  
+            if line.is_a?(Engine::Action::Message)
               sender = line.entity.name || line.user
               line = "#{sender}: #{line.message}"
             end
-            log_text += time ? "#{time} #{line}\n" : "#{line}\n"
-          end  
+            log_text.push(time ? "#{time} #{line}" : line)
```
Comment (tobymao) [inline assets/app/view/game/game_log.rb:243]: log_text << not push

**Pull request merged**